gnovm/stdlibs/std/emit_event.go

```diff
@@ -0,0 +1,77 @@
+package std
+
+// ref: https://github.com/gnolang/gno/pull/853
+
+import (
+	"encoding/json"
+	"errors"
+
+	gno "github.com/gnolang/gno/gnovm/pkg/gnolang"
+	abci "github.com/gnolang/gno/tm2/pkg/bft/abci/types"
+)
+
+func X_emit(m *gno.Machine, typ string, attrs []string) {
+	eventAttrs, err := attrKeysAndValues(attrs)
+	if err != nil {
+		m.Panic(typedString(err.Error()))
+	}
+
+	pkgPath := CurrentRealmPath(m)
+	fnIdent := getPrevFunctionNameFromTarget(m, "Emit")
+	timestamp := getTimestamp(m)
+
+	evtstr := NewGnoEventString(typ, pkgPath, fnIdent, timestamp, eventAttrs...)
+	ctx := m.Context.(ExecContext)
+	ctx.EventLogger.EmitEvent(evtstr)
+}
+
+func attrKeysAndValues(attrs []string) ([]gnoEventAttribute, error) {
+	attrLen := len(attrs)
+	if attrLen%2 != 0 {
+		return nil, errors.New("odd number of attributes. cannot create key-value pairs")
```

Comment:
```suggestion
		return nil, errors.New("cannot pair attributes due to odd count")
```

avoiding using the dot in this wrapable error.